.github/workflows/automatic-cherry-pick.yaml

```diff
@@ -0,0 +1,246 @@
+      status: ${{ steps.report.outputs.status }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          # We need the history of all branches for the sanity checks
+          fetch-depth: 0
+
+      - name: Parse Command
+        # TODO: make this step into a separate job, so that we can handle more than one command per comment
+        id: command
+        shell: bash
+        run: |
+          set -f
+          command_string=$(echo "${{ github.event.comment.body }}" | grep -m 1 cherry-pick)
+          echo "command=$command_string" >> $GITHUB_OUTPUT
+          command=($command_string)
+          command=("${command[@]:1}")
+
+          has_separator=false
+          arg_type=hash
+          commits=()
+          branches=()
+          for token in ${command[@]}; do
+            if [[ "$token" == "into" ]]; then
+              arg_type=branch
+              has_separator=true
+            else
+              if [[ "$arg_type" == "hash" ]]; then
+                commits+=( $token )
+              elif [[ "$arg_type" == "branch" ]]; then
+                branches+=( $token )
+              fi
+            fi
+          done
+
+          # Check command correctness
+          errors=""
+          if [[ "$has_separator" = false ]]; then
+            errors="$errors - the command is missing the \\\`into\\\` separator
+          "
+          fi
+          if [[ -z "$commits" ]]; then
+            errors="$errors - no list of commits to cherry-pick was provided
+          "
+          fi
+          if [[ -z "$branches" ]]; then
+            errors="$errors - no list of target branches was provided
+          "
+          fi
+          if [[ -n "$errors" ]]; then
+            errors="Incorrect cherry-pick command:
+          $errors"
+            printf "ERROR_MSG<<EOF\n%s\nEOF" "$errors" >> $GITHUB_ENV
+          else
+            # Output lists of commits and branches
+            echo "commits=${commits[@]}" >> $GITHUB_OUTPUT
+            echo "branches=${branches[@]}" >> $GITHUB_OUTPUT
+
+            # We also output the list of branches as json, so they can be used to generate a matrix for the next job
+            echo "branch_matrix=$(jq -cn '$ARGS.positional' --args -- "${branches[@]}")" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Check PR Status
+        if: env.ERROR_MSG == ''
+        run: |
+          # Check if the PR has been merged
+          if [[ -z "${{ github.event.issue.pull_request.merged_at }}" ]]; then
+            echo "ERROR_MSG=Pull request has not been merged yet. Cannot cherry-pick commits." >> $GITHUB_ENV
+          fi
+
+      - name: Check commits
+        if: env.ERROR_MSG == ''
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          # Check that the commits to cherry-pick are actually part of this PRs target branch
+          target=$(gh api repos/{owner}/{repo}/pulls/${{ github.event.issue.number }} -q .base.ref)
```

Comment:
Didn't know that the `{owner}/{repo}` bit is filled in by `gh`! Very cool. Was about to suggest replacing it with `${{ github.repository }}` :grin:

Reply:
Syntax sugar is so sweet!

Reply:
Yeah, `gh api` has some very cool capabilities. The `-q` flag is priceless.